### ISSUE TITLE
The apiserver usage cpu high beause istiod constant reconciling gateway resource

### ISSUE THREAD

**Issue by lklkxcxc** [istio/istio#48269]
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

### Bug Description

The apiserver usage cpu high beause istiod constant reconciling gateway resource but no change gateway resource and httproute. The logs output very often: 
```
2023-12-11T11:24:22.722045Z	info	ads	Push debounce stable[17491] 2 for config Gateway/gateway-system/ops-ingress: 107.297824ms since last change, 147.467401ms since last push, full=true
2023-12-11T11:24:22.736178Z	info	ads	XDS: Pushing:2023-12-11T11:24:22Z/17491 Services:333 ConnectedEndpoints:3 Version:2023-12-11T11:24:22Z/17491
2023-12-11T11:24:22.741718Z	info	ads	CDS: PUSH for node:ops-ingress-655cb7ddf5-ggqst.gateway-system resources:815 size:818.9kB cached:814/814
2023-12-11T11:24:22.745864Z	info	ads	EDS: PUSH for node:ops-ingress-655cb7ddf5-ggqst.gateway-system resources:814 size:237.0kB empty:0 cached:814/814
2023-12-11T11:24:22.746710Z	info	ads	LDS: PUSH for node:ops-ingress-655cb7ddf5-ggqst.gateway-system resources:1 size:4.6kB
2023-12-11T11:24:22.746912Z	info	ads	RDS: PUSH for node:ops-ingress-655cb7ddf5-ggqst.gateway-system resources:1 size:1.8kB cached:0/0
2023-12-11T11:24:22.747204Z	info	gateway	reconciling	gateway=gateway-system/ops-ingress
2023-12-11T11:24:22.756017Z	info	gateway	service updated	gateway=gateway-system/ops-ingress
2023-12-11T11:24:22.761606Z	info	ads	CDS: PUSH for node:istio-ingressgateway-75b6d595bc-92xwr.istio-system resources:815 size:818.9kB cached:814/814
2023-12-11T11:24:22.761985Z	info	gateway	deployment updated	gateway=gateway-system/ops-ingress
2023-12-11T11:24:22.767779Z	info	ads	EDS: PUSH for node:istio-ingressgateway-75b6d595bc-92xwr.istio-system resources:814 size:237.0kB empty:0 cached:814/814
2023-12-11T11:24:22.772075Z	info	ads	LDS: PUSH for node:istio-ingressgateway-75b6d595bc-92xwr.istio-system resources:1 size:4.6kB
2023-12-11T11:24:22.782445Z	info	gateway	gateway updated	gateway=gateway-system/ops-ingress
2023-12-11T11:24:22.782491Z	info	gateway	reconciling	gateway=gateway-system/ops-ingress
2023-12-11T11:24:22.789785Z	info	gateway	service updated	gateway=gateway-system/ops-ingress
2023-12-11T11:24:22.796118Z	info	gateway	deployment updated	gateway=gateway-system/ops-ingress
2023-12-11T11:24:22.803633Z	info	ads	RDS: PUSH for node:istio-ingressgateway-75b6d595bc-2tnk9.istio-system resources:1 size:17.6MB cached:0/0
2023-12-11T11:24:22.810943Z	info	gateway	gateway updated	gateway=gateway-system/ops-ingress
2023-12-11T11:24:22.883053Z	info	ads	Push debounce stable[17492] 2 for config Gateway/gateway-system/ops-ingress: 101.063899ms since last change, 135.689848ms since last push, full=true
```
When delete gateway “gateway-system/ops-ingress” return to normal

### Version

```Text
istio version:1.16.3
kubernetes version:  v1.19.9
helm version: v3.12.3+g3a31588
```

**Comment by zirain** [comment]
1.16 is EOL, can you upgrade or try to reproduce with latest version?

**Comment by lklkxcxc** [comment]
why resourceVersion in flux：
```
[root@k8s-master-002 ~]# kubectl -n gateway-system get gateway ops-ingress -o yaml|grep resourceVersion
  resourceVersion: "171038828"
[root@k8s-master-002 ~]# kubectl -n gateway-system get gateway ops-ingress -o yaml|grep resourceVersion
  resourceVersion: "171038860"
[root@ningxia-k8s-master-002 ~]# kubectl -n gateway-system get gateway ops-ingress -o yaml|grep resourceVersion
  resourceVersion: "171038898"
[root@k8s-master-002 ~]# kubectl -n gateway-system get gateway ops-ingress -o yaml|grep resourceVersion
  resourceVersion: "171038933"
[root@k8s-master-002 ~]# kubectl -n gateway-system get gateway ops-ingress -o yaml|grep resourceVersion
  resourceVersion: "171038968"
[root@k8s-master-002 ~]# kubectl -n gateway-system get gateway ops-ingress -o yaml|grep resourceVersion
  resourceVersion: "171039008"
[root@k8s-master-002 ~]# kubectl -n gateway-system get gateway ops-ingress -o yaml|grep resourceVersion
  resourceVersion: "171039039"
[root@k8s-master-002 ~]# kubectl -n gateway-system get gateway ops-ingress -o yaml|grep resourceVersion
  resourceVersion: "171039086"
[root@k8s-master-002 ~]# kubectl -n gateway-system get gateway ops-ingress -o yaml|grep resourceVersion
```

**Comment by dddddai** [comment]
I think this should be fixed by https://github.com/istio/istio/pull/44014, which is shipped after 1.18

**Comment by lklkxcxc** [comment]
@dddddai thanks
